frame/bridge/eth/offchain/src/tests.rs

```diff
@@ -6,55 +6,33 @@ use frame_support::{assert_noop, assert_ok};
-			.unwrap();
-	assert_eq!(
-		result_part,
-		br#""eth_header":{eth-content},"proof":[proof-content]"# as &[u8]
-	);
-	let eth_header_part =
-		EthOffchain::extract_from_json_str(result_part, b"eth_header" as &[u8]).unwrap();
-	assert_eq!(eth_header_part, br#"eth-content"# as &[u8]);
-	let proof_header_part =
-		EthOffchain::extract_from_json_str(result_part, b"proof" as &[u8]).unwrap();
-	assert_eq!(proof_header_part, br#"proof-content"# as &[u8]);
+	let result_part = extract_from_json_str(&SUPPOSED_SHADOW_FAKE_RESPONSE[..], b"result" as &[u8]).unwrap();
+	assert_eq!(result_part, br#""eth_header":{eth-content},"proof":[proof-content]"# as &[u8]);
+	let eth_header_part = extract_from_json_str(result_part, b"eth_header" as &[u8]).unwrap();
+	assert_eq!(eth_header_part , br#"eth-content"# as &[u8]);
+	let proof_header_part = extract_from_json_str(result_part, b"proof" as &[u8]).unwrap();
+	assert_eq!(proof_header_part , br#"proof-content"# as &[u8]);
 }
 
 /// Extract value from JSON response with not alphabetical order
 #[test]
 fn test_extract_value_from_non_order_json_response() {
-	let result_part = EthOffchain::extract_from_json_str(
-		&SUPPOSED_SHADOW_NON_ORDER_RESPONSE[..],
-		b"result" as &[u8],
-	)
-	.unwrap();
-	assert_eq!(
-		result_part,
-		br#""proof":[proof-content],"eth_header":{eth-content}"# as &[u8]
-	);
-	let eth_header_part =
-		EthOffchain::extract_from_json_str(result_part, b"eth_header" as &[u8]).unwrap();
-	assert_eq!(eth_header_part, br#"eth-content"# as &[u8]);
-	let proof_header_part =
-		EthOffchain::extract_from_json_str(result_part, b"proof" as &[u8]).unwrap();
-	assert_eq!(proof_header_part, br#"proof-content"# as &[u8]);
+	let result_part = extract_from_json_str(&SUPPOSED_SHADOW_NON_ORDER_RESPONSE[..], b"result" as &[u8]).unwrap();
+	assert_eq!(result_part, br#""proof":[proof-content],"eth_header":{eth-content}"# as &[u8]);
+	let eth_header_part = extract_from_json_str(result_part, b"eth_header" as &[u8]).unwrap();
+	assert_eq!(eth_header_part , br#"eth-content"# as &[u8]);
+	let proof_header_part = extract_from_json_str(result_part, b"proof" as &[u8]).unwrap();
+	assert_eq!(proof_header_part , br#"proof-content"# as &[u8]);
 }
 
 /// Basice JSON response handle
 #[test]
 fn test_build_eth_header_from_json_response() {
-	let eth_header_part = EthOffchain::extract_from_json_str(
-		&SUPPOSED_SHADOW_JSON_RESPONSE[..],
-		b"eth_header" as &[u8],
-	)
-	.unwrap_or_default();
+	let eth_header_part = extract_from_json_str(&SUPPOSED_SHADOW_JSON_RESPONSE[..], b"eth_header" as &[u8]).unwrap_or_default();
 	let header = EthHeader::from_str_unchecked(from_utf8(eth_header_part).unwrap_or_default());
 	assert_eq!(header.hash.unwrap(), header.re_compute_hash());
 
-	let proof_part =
-		EthOffchain::extract_from_json_str(&SUPPOSED_SHADOW_JSON_RESPONSE[..], b"proof" as &[u8])
-			.unwrap_or_default();
+	let proof_part = extract_from_json_str(&SUPPOSED_SHADOW_JSON_RESPONSE[..], b"proof" as &[u8]).unwrap_or_default();
```

Comment:
Supposed to use serde_json instead, this is why I was requesting @AurevoirXavier that "extract_from_json_str" should limited to eth-primitives tests.

I not keen using *two* different tests datas in pallet tests here.

I will create a separate issue for this refactoring later.